### DISCUSSION TITLE
Create an exe with pyinstaller (Windows 10)

### DISCUSSION THREAD

**Discussion by B3rthi** [mathoudebine/turing-smart-screen-python#385] [General]
First @mathoudebine thank you ! I didn't want to use the executable coming with my recently purchased screen but I'm a complete beginner in coding and your project is very well documented and clearly explained, and I have now a perfectly functionnal monitor

It may be obvious for some but I created an executable of configure.py because I wanted to quickly start my screen only when gaming. If it can help a beginner like me :
- Install pyinstaller module and go into this project folder where configure.py is located (avoid spaces in the name of this folder)
- Run `pyinstaller configure.py --onefile --windowed --uac-admin --collect-data sv_ttk` (remove --windowed if you want to keep the cmd window)
- A new folder dist has been created with the exe in the project folder, copy the exe into the main folder just above configure.py
- Create a shortcut for the exe or anything else !

For main.py, it was working fine but I had a warning in the console asking for babel.numbers so I added it :
`pyinstaller main.py --onefile --windowed --uac-admin --collect-data sv_ttk --hidden-import "babel.numbers"`

Everything seems to be working on Windows 10, not sure about W11, Linux ...

**Comment by mathoudebine** [comment]
Hi! Thanks for letting me know, it could be useful to have .exe instead of .py if someday we develop a Windows installer 👍

**Comment by jimaiwr** [comment]
Thank you so much! was trying to do this but it kept opening cmd up couldn't close it without closing the program too. This is eactly what I wanted and easy to do thanks!

**Comment by CodeAsm** [comment]
Altho this seemingly helped a bit with my atempt. (https://github.com/pyinstaller/pyinstaller/blob/develop/.github/ISSUE_TEMPLATE/antivirus.md#reporting-false-positives-to-av-vendors) 
now I get a .NET fault:

```1/25/2024 3:01:16 PM [DEBUG] HW revision: A
1/25/2024 3:01:16 PM [DEBUG] Static COM port: COM6
System.NotSupportedException: An attempt was made to load an assembly from a network location which would have caused the assembly to be sandboxed in previous versions of the .NET Framework. This release of the .NET Framework does not enable CAS policy by default, so this load may be dangerous. If this load is not intended to sandbox the assembly, please enable the loadFromRemoteSources switch. See http://go.microsoft.com/fwlink/?LinkId=155569 for more information.

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:\Users\CodeAsm\Downloads\IPSMonitor\turing-smart-screen-python-3.3.4\main.py", line 64, in <module>
    import library.scheduler as scheduler
  File "C:\Users\CodeAsm\Downloads\IPSMonitor\turing-smart-screen-python-3.3.4\library\scheduler.py", line 28, in <module>
    import library.stats as stats
  File "C:\Users\CodeAsm\Downloads\IPSMonitor\turing-smart-screen-python-3.3.4\library\stats.py", line 63, in <module>
    import library.sensors.sensors_librehardwaremonitor as sensors
  File "C:\Users\CodeAsm\Downloads\IPSMonitor\turing-smart-screen-python-3.3.4\library\sensors\sensors_librehardwaremonitor.py", line 40, in <module>
    clr.AddReference(lhm_dll)
System.IO.FileLoadException: Could not load file or assembly 'file:///C:\Users\CodeAsm\Downloads\IPSMonitor\turing-smart-screen-python-3.3.4\external\LibreHardwareMonitor\LibreHardwareMonitorLib.dll' or one of its dependencies. Operation is not supported. (Exception from HRESULT: 0x80131515)
File name: 'file:///C:\Users\CodeAsm\Downloads\IPSMonitor\turing-smart-screen-python-3.3.4\external\LibreHardwareMonitor\LibreHardwareMonitorLib.dll' ---> System.NotSupportedException: An attempt was made to load an assembly from a network location which would have caused the assembly to be sandboxed in previous versions of the .NET Framework. This release of the .NET Framework does not enable CAS policy by default, so this load may be dangerous. If this load is not intended to sandbox the assembly, please enable the loadFromRemoteSources switch. See http://go.microsoft.com/fwlink/?LinkId=155569 for more information.
   at System.Reflection.RuntimeAssembly._nLoad(AssemblyName fileName, String codeBase, Evidence assemblySecurity, RuntimeAssembly locationHint, StackCrawlMark& stackMark, IntPtr pPrivHostBinder, Boolean throwOnFileNotFound, Boolean forIntrospection, Boolean suppressSecurityChecks)
   at System.Reflection.RuntimeAssembly.InternalLoadAssemblyName(AssemblyName assemblyRef, Evidence assemblySecurity, RuntimeAssembly reqAssembly, StackCrawlMark& stackMark, IntPtr pPrivHostBinder, Boolean throwOnFileNotFound, Boolean forIntrospection, Boolean suppressSecurityChecks)
   at System.Reflection.RuntimeAssembly.InternalLoadFrom(String assemblyFile, Evidence securityEvidence, Byte[] hashValue, AssemblyHashAlgorithm hashAlgorithm, Boolean forIntrospection, Boolean suppressSecurityChecks, StackCrawlMark& stackMark)
   at System.Reflection.Assembly.LoadFrom(String assemblyFile)
   at Python.Runtime.AssemblyManager.LoadAssemblyFullPath(String name)
   at Python.Runtime.CLRModule.AddReference(String name)
```

Unless someone can tell me how to fix this, Im done with windows 11. I go back using Arch where stuff just works. (also why i left MS windows 10 years ago... the ever changing ways of doing crap  without any documents and 100 ways of "update your drivers, reboot")
this should be a seperate issue, but if you reach this error too, just install mac or linux, cause simple-program.py just works, this just another example of why things in Linux is better tested and working out of the box. Mathoudebine's project works fine on my Debian Raspberry Pi and Arch laptop, desktop and Ubuntu Netbook.

**Reply by mathoudebine** [reply]
For this issue you can have a look at this Troubleshooting page: https://github.com/mathoudebine/turing-smart-screen-python/wiki/Troubleshooting#windows-cannot-load-librehardwaremonitordll-loadfromremotesources

**Comment by Hecrot** [comment]
> First @mathoudebine thank you ! I didn't want to use the executable coming with my recently purchased screen but I'm a complete beginner in coding and your project is very well documented and clearly explained, and I have now a perfectly functionnal monitor
> 
> It may be obvious for some but I created an executable of configure.py because I wanted to quickly start my screen only when gaming. If it can help a beginner like me :
> 
> * Install pyinstaller module and go into this project folder where configure.py is located (avoid spaces in the name of this folder)
> * Run `pyinstaller configure.py --onefile --windowed --uac-admin --collect-data sv_ttk` (remove --windowed if you want to keep the cmd window)
> * A new folder dist has been created with the exe in the project folder, copy the exe into the main folder just above configure.py
> * Create a shortcut for the exe or anything else !
> 
> For main.py, it was working fine but I had a warning in the console asking for babel.numbers so I added it : `pyinstaller main.py --onefile --windowed --uac-admin --collect-data sv_ttk --hidden-import "babel.numbers"`
> 
> Everything seems to be working on Windows 10, not sure about W11, Linux ...

Hello, first of all, excuse my English (Google Translate), I have tried to replicate what you indicate but I always get the same error: "pyinstaller" is not recognized as an internal or external command,
program or executable batch file.

I guess I'm doing something wrong, this is my first approach to Python because of the screen.

I managed to get it to start using a simple .bat and a scheduled task, it is probably the least beautiful way to do it but for now it works.


But the truth is I would like to be able to create the .exe as you indicate.

Thanks in advance.

**Reply by B3rthi** [reply]
Hello !

Did you installed pyinstaller first ? You need to run into CMD :
`pip install pyinstaller`

**Reply by Hecrot** [reply]
Yes, install pyinstaller first.

**Reply by Hecrot** [reply]
C:\Windows\System32>pip install pyinstaller
Requirement already satisfied: pyinstaller in c:\users\hecrot\appdata\roaming\python\python312\site-packages (6.3.0)
Requirement already satisfied: setuptools>=42.0.0 in c:\users\hecrot\appdata\roaming\python\python312\site-packages (from pyinstaller) (69.0.3)
Requirement already satisfied: altgraph in c:\users\hecrot\appdata\roaming\python\python312\site-packages (from pyinstaller) (0.17.4)
Requirement already satisfied: pyinstaller-hooks-contrib>=2021.4 in c:\users\hecrot\appdata\roaming\python\python312\site-packages (from pyinstaller) (2024.0)
Requirement already satisfied: packaging>=22.0 in c:\users\hecrot\appdata\roaming\python\python312\site-packages (from pyinstaller) (23.2)
Requirement already satisfied: pefile>=2022.5.30 in c:\users\hecrot\appdata\roaming\python\python312\site-packages (from pyinstaller) (2023.2.7)
Requirement already satisfied: pywin32-ctypes>=0.2.1 in c:\users\hecrot\appdata\roaming\python\python312\site-packages (from pyinstaller) (0.2.2)
